grails-bom/build.gradle

```diff
@@ -1,108 +1,124 @@
-apply plugin: 'maven-publish'
+plugins {
+    id 'java-platform'
+    id 'maven-publish'
+}
 
-publishing {
-    publications {
-        maven(MavenPublication) {
-            pom.withXml {
-                def xml = asNode()
-                def plugins = new Properties()
-                plugins.load(new StringReader(new File("$projectDir/plugins.properties").text))
+javaPlatform {
+    allowDependencies()
+}
 
-                def profiles = new Properties()
-                profiles.load(new StringReader(new File("$projectDir/profiles.properties").text))
+def plugins = new Properties()
+plugins.load(new StringReader(new File("$projectDir/plugins.properties").text))
 
-                xml.children().last() + pomInfo
-                xml.children().last() + {
-                    def mkp = delegate
+def profiles = new Properties()
+profiles.load(new StringReader(new File("$projectDir/profiles.properties").text))
 
-                    mkp.properties {
-                        for (dep in plugins) {
-                            String version = dep.value
-                            if (!isBuildSnapshot && version.endsWith("-SNAPSHOT")) {
-                                throw new RuntimeException("Cannot have a snapshot dependency on a plugin [$dep.key] for a release!")
-                            }
-                            mkp."${dep.key}.version"(version)
-                        }
+ext {
+    // When making changes in the dependencyVersions, remember to also update the Grails BOM Documentation:
+    // https://docs.grails.org/snapshot/ref/Dependency%20Versions/Grails%20BOM.html
+    dependencyVersions = """\
+        com.fasterxml.jackson.core:jackson-databind::$jacksonVersion
+        com.github.ben-manes.caffeine:caffeine::$caffeineVersion
+        com.github.javaparser:javaparser-core::$javaParserCoreVersion
+        com.h2database:h2::$h2Version
+        commons-codec:commons-codec::$commonsCodecVersion
+        hsqldb:hsqldb::$hsqldbVesion
```

Comment:
Are we still using `hsqldb`?

Reply:
I am not sure. I took it from here. Can it be removed?

```gradle
    if (subproject.name =~ /^(grails-plugin-datasource|grails-test-suite)/) {
        dependencies {
            testImplementation "hsqldb:hsqldb"
            testImplementation "com.h2database:h2"
        }
    }
```

Reply:
I think `hsqldb` was replaced with `h2`, and its just a remnant from that.
https://grails.github.io/grails2-doc/2.1.1/guide/upgradingFromPreviousVersionsOfGrails.html

Reply:
Not sure what this does:

https://github.com/grails/grails-core/blob/0342bd126723d9f77f88f715c3fdfa2a5a1c0d5f/grails-plugin-datasource/src/main/groovy/org/grails/plugins/datasource/EmbeddedDatabaseShutdownHook.groovy#L71-L82

Reply:
It looks like it is used for adding shutdown hooks for embedded in-memory databases.
I my view, it does not really impact removal of `hsqldb` from dependencies and `bom`.

Reply:
removed